functests/dpdk/dpdk.go

```diff
@@ -107,21 +109,29 @@ var _ = Describe("dpdk", func() {
 			if !discoverySuccessful {
 				return
 			}
-			performanceProfiles = filterPerformanceProfilesWithAvailableNodes(performanceProfiles, nodeSelector)
-			if len(performanceProfiles) == 0 {
-				discoverySuccessful, discoveryFailedReason = false, "Can not run tests in discovery mode. Failed to find a matching node"
+
+			profile, discoveredResourceName, sriovDevice, err := findPerformanceProfileAndPolicyWithAvailableNodes(performanceProfiles, nodeSelector)
+			resourceName = discoveredResourceName
+			if err != nil {
+				discoverySuccessful, discoveryFailedReason = false, "Can not run tests in discovery mode. Failed to discover required resources."
 				return
 			}
-			// TODO: use the first matching profile for now. When sriov discovery is added, a profile with
-			// an available sriov node should be chosen
-			nodeSelector = performanceProfiles[0].Spec.NodeSelector
+			nodeSelector = nodes.NodesSelectorUnion(nodeSelector, profile.Spec.NodeSelector)
+
+			validSriovNetwork, err := ValidateSriovNetwork(resourceName)
```

Comment:
Why not creating it always as we do in sriov? The get looks for a specific name that won't be there in discovery mode.

Reply:
This is the original code (moved from [here](https://github.com/openshift-kni/cnf-features-deploy/pull/276/files#diff-3330f47928cf477b3ddae2071f543f66L404) ).

Reply:
I see, nut now that we are more explicit about discovery / not discovery, when running in non discovery mode the network won't be there. I'd make that part simpler and just create the network